source/platform/file_formats/ncch.hpp

```diff
@@ -466,14 +497,45 @@ struct NCSDHeader {
             (MediaUnit32, size)
         );
     };
+    
+    struct CartHeader {
+        BOOST_HANA_DEFINE_STRUCT(CartHeader,
+            (std::array<uint8_t, 32>, exheader_sha256),
+            (uint32_t, additional_header_size),
+            (uint32_t, sector0_offset),
+            (uint64_t, partition_flags),
+            (std::array<uint64_t, 8>, partition_id_table),
```

Comment:
Could you add a brief inline-comment for this? Something like "Non-zero if a partition of the given id exists"